### PR TITLE
Registered check in

### PR DESCRIPTION
**Pull request by KenG98**
Check in capabilities at website/check-in url. It requires the person checking in be an admin to access that page. Also merged the backup scripts into this, just to get em on the master branch.

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -31,7 +31,10 @@ const userSchema = mongoose.Schema({
   additional: String,
   status: String, // Possible values so far: {'Pending', 'Not Attending'}
   notAttendingReason: String,
-  dataSharingAgreed: Boolean
+  dataSharingAgreed: Boolean,
+  checkedIn: {
+    required: false
```
Comment (a--hoang) [inline models/user.js]: Can we make this a boolean field too pls

File: public/js/admin-check-in.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+function httpGetAsync(theUrl, callback){
+  var xmlHttp = new XMLHttpRequest();
+  xmlHttp.onreadystatechange = function() { 
+    if (xmlHttp.readyState == 4 && xmlHttp.status == 200)
```
Comment (a--hoang) [inline public/js/admin-check-in.js:4]: le sigh. hoping to add webpack + promise based library like superagent so we dont have to do this, but i'll take it

File: routes/index.js [hunk 3]
```diff
@@ -266,6 +266,32 @@ router.get('/changeStatus/:userid/:stat', isLoggedIn, function(req, res) {
   })
 })
 
+router.get('/check-in', isLoggedIn, function (req, res, next) {
+  if (!adminFunc.isAdmin(req.user.local.email)) {
+    res.redirect('/')
+    return
```
Comment (a--hoang) [inline routes/index.js]: return on same line if possible

File: routes/index.js [hunk 4]
```diff
@@ -266,6 +266,32 @@ router.get('/changeStatus/:userid/:stat', isLoggedIn, function(req, res) {
   })
 })
 
+router.get('/check-in', isLoggedIn, function (req, res, next) {
+  if (!adminFunc.isAdmin(req.user.local.email)) {
```
Comment (a--hoang) [inline routes/index.js]: we should break out this method into a isAdmin middleware (like isLoggedIn but for admins only

File: routes/index.js [hunk 5]
```diff
@@ -266,6 +266,32 @@ router.get('/changeStatus/:userid/:stat', isLoggedIn, function(req, res) {
...
+    res.redirect('/')
+    return
+  }
+  res.render('pages/admin-check-in.ejs')
+})
+
+router.get('/checkUserIn/:userid', isLoggedIn, function(req, res) {
+  if (!adminFunc.isAdmin(req.user.local.email)) {
```
Comment (a--hoang) [inline routes/index.js]: same as above

File: routes/index.js [hunk 6]
```diff
@@ -266,6 +266,32 @@ router.get('/changeStatus/:userid/:stat', isLoggedIn, function(req, res) {
...
+    return res.redirect('/')
+  }
+  adminFunc.checkUserIn(req.params.userid, function() {
+    return res.sendStatus(200)
+  })
+})
+
+router.get('/unCheckUserin/:userid', isLoggedIn, function(req, res) {
+  if (!adminFunc.isAdmin(req.user.local.email)) {
```
Comment (a--hoang) [inline routes/index.js]: one more

File: scripts/backupDB.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# run this script on YOUR OWN computer
+
+# runs mongodump on the production server
+ssh root@159.203.171.222 "mongodump"
```
Comment (a--hoang) [inline scripts/backupDB.sh]: well, so much for keeping the server ip secret 🔒

Comment (a--hoang) [reply scripts/backupDB.sh]: maybe a bash variable?

**Review by a--hoang — CHANGES_REQUESTED** [review]
@KenG98 lookin' good bro, your node skills got way better since the days of brandon. a few changes pls

File: public/js/admin-check-in.js (hunk 2 above)
Comment (anna1go) [reply public/js/admin-check-in.js:4]: Yea https://github.com/Bostonhacks/headstart/issues/57 
Ken and I had this discussion over messenger; possibly for next year

File: scripts/backupDB.sh (hunk 7 above)
Comment (anna1go) [reply scripts/backupDB.sh]: Yeah, def take this out

File: config/admin.js [hunk 8]
```diff
@@ -27,5 +27,15 @@ module.exports = {
   // returns true if the email given is an admin email
   isAdmin: function(emailAddr) {
     return process.env.ADMIN_EMAILS.includes(emailAddr)
+  },
+  checkUserIn: function(userid, next) {
+    User.update({_id: userid}, {$set: {checkedIn: new Date()}}, function(err, raw){
+      return next()
+    })
+  }, 
+  unCheckUserIn: function(userid, next) {
```
Comment (anna1go) [inline config/admin.js]: the name is kind of awkward.... what is the intended purpose of this method?

File: public/js/admin-check-in.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  xmlHttp.send(null);
+}
+
+var app; 
+
+httpGetAsync('/admin-all-registrants', function(result) {
+  app = new Vue({
+    el: '#app',
+    data: {
+      users: JSON.parse(result).map(u => {
```
Comment (anna1go) [inline public/js/admin-check-in.js]: I am just wondering, how come an arrow function is allowed? Is this bundled with the Vue library?

**Review by anna1go — CHANGES_REQUESTED** [review]
I typically put semicolons at the end of my lines, but we don't have a standard for this code, which typically varies from an org to an org, so that's fine for now. 

Awesome job, just a couple of small things. Def if possible, isolate the IPs into a separate file

**Review by anna1go — APPROVED** [review]

**Comment by anna1go** [comment]
Thanks!

**Comment by KenG98** [comment]
New "isAdmin" middleware, it's similar to "isLoggedIn". I also made most of the above changes. This is good stuff, I'm learning a lot. LMK when it's good to merge. Thanks for the reviews!

**Review by a--hoang — APPROVED** [review]

**Pull request merged**